sktime/classification/deep_learning/base/_base_torch.py

```diff
@@ -0,0 +1,314 @@
+        "SGD": torch.optim.SGD,
+    }
+else:
+
+    class Dataset:
+        """Dummy class if torch is unavailable."""
+
+
+class BaseDeepClassifierPytorch(BaseClassifier):
+    """Abstract base class for the Pytorch neural network classifiers."""
+
+    _tags = {
+        "authors": ["geetu040", "RecreationalMath"],
+        "maintainers": ["geetu040", "RecreationalMath"],
+        "python_dependencies": ["torch"],
+        "X_inner_mtype": "numpy3D",
+        "y_inner_mtype": "numpy1D",
+        "capability:multivariate": True,
+        "capability:multioutput": False,
+    }
+
+    def __init__(
+        self,
+        num_epochs=16,
+        batch_size=8,
+        criterion=None,
+        criterion_kwargs=None,
+        optimizer=None,
+        optimizer_kwargs=None,
+        lr=0.001,
+        verbose=True,
+        random_state=None,
+    ):
+        self.num_epochs = num_epochs
+        self.batch_size = batch_size
+        self.criterion = criterion
+        self.criterion_kwargs = criterion_kwargs
+        self.optimizer = optimizer
+        self.optimizer_kwargs = optimizer_kwargs
+        self.lr = lr
+        self.verbose = verbose
+        self.random_state = random_state
+
+        if self.random_state is not None:
+            if _check_soft_dependencies("torch", severity="none"):
+                torch.manual_seed(self.random_state)
+
+        # use this when y has str
+        self.label_encoder = None
+        super().__init__()
+
+        # instantiate optimizers
+        self.optimizers = OPTIMIZERS
+
+    def _fit(self, X, y):
+        y = self._encode_y(y)
+
+        self.network = self._build_network(X, y)
+
+        self._criterion = self._instantiate_criterion()
+        self._optimizer = self._instantiate_optimizer()
+
+        dataloader = self._build_dataloader(X, y)
+
+        self.network.train()
+        for epoch in range(self.num_epochs):
+            self._run_epoch(epoch, dataloader)
+
+    def _run_epoch(self, epoch, dataloader):
+        losses = []
+        for inputs, outputs in dataloader:
+            y_pred = self.network(**inputs)
+            loss = self._criterion(y_pred, outputs)
+            self._optimizer.zero_grad()
+            loss.backward()
+            self._optimizer.step()
+            losses.append(loss.item())
+        if self.verbose:
+            print(f"Epoch {epoch + 1}: Loss: {np.average(losses)}")
+
+    def _instantiate_optimizer(self):
+        if self.optimizer:
+            if self.optimizer in self.optimizers.keys():
+                if self.optimizer_kwargs:
+                    return self.optimizers[self.optimizer](
+                        self.network.parameters(), lr=self.lr, **self.optimizer_kwargs
+                    )
+                else:
+                    return self.optimizers[self.optimizer](
+                        self.network.parameters(), lr=self.lr
+                    )
+            else:
+                raise TypeError(
```

Comment:
I think it should be possible to pass the optimizer directly. String shorthands are ok, but why do we not allow passing the optimizer instance directly?